xlml/apis/task.py

```diff
@@ -436,7 +438,7 @@ def launch_workload(self, workload_id: str, gcs_path: str) -> DAGNode:
       run_workload >> wait_for_workload_start
       return group
 
-  def post_process(self) -> DAGNode:
+  def post_process(self, result_location: Optional[str] = None) -> DAGNode:
```

Comment:
I think you mean the following:
```
with TaskGroup(...): 
    gcs_location = = name_format.generate_gcs_folder_location(...)
    run_model = self.run_model(gcs_location)
    self.post_process(gcs_location)
```
This is how I usually pass the gcs_location.

Reply:
Yeah that sounds good. @raymondzouu could we pull the `name_format.generate_gcs_folder_location` up into `run` from `run_model`? So `run` will look like:

```python
with TaskGroup(...):
  gcs_location = gcs_location or name_format.generate_gcs_folder_location(...)
  self.run_model(gcs_location) >> self.post_process(gcs_location)
```

And ignore my other comment about updating the return value on L410. Thanks!

Reply:
Thanks for pointing this out. I think it would be better to keep the `generate_gcs_folder_location` logic inside `run_model` so it can be ran along side `generate_workload_id` and will follow a similar structure to `TpuQueuedResourceTask`. Instead, I changed it so that `run_model` just returns the generated `gcs_path` and passes it to `post_process`. Wdyt?

Reply:
I think it's a bit more clear how the GCS path is generated if we pull it to the top-level of `run`, but I'm okay with either approach.

Reply:
Thanks all! I am fine with current implementation that gcs path is generate inside of `run_with_run_name_generation` as the name indicated. I understand currently, we have that outside for run() to support chained tests.